lib/global_id/locator.rb

```diff
@@ -106,7 +106,9 @@ def use(app, locator = nil, &locator_block)
 
       private
         def locator_for(gid)
-          @locators.fetch(normalize_app(gid.app)) { default_locator }
+          @locators.fetch(normalize_app(gid.app)) do
+            gid.respond_to?(:unscoped) ? ACTIVE_RECORD_LOCATOR : DEFAULT_LOCATOR
```

Comment:
Yep I missed that this methods is tight to STI. I think assuming that is an AR if it responds to `:unscoped` is wrong and will bring trouble in the future.